File: analysis/read_data.R
# Read the synced Seed Production Log from your Google Sheet into R.
#
# 1. In the Google Sheet: File -> Share -> Publish to web -> (Entire document,
#    CSV) OR just make the sheet viewable, then use the export URL below.
# 2. Replace SHEET_ID with your sheet's id (the long string in its URL:
#    https://docs.google.com/spreadsheets/d/SHEET_ID/edit )

sheet_id <- "1bfQs9LS-zign0eLmWYA8JfJ3LKBcMF7PDRLgtbMbZKU"
url <- paste0("https://docs.google.com/spreadsheets/d/", sheet_id, "/export?format=csv")

df <- read.csv(url, stringsAsFactors = FALSE, check.names = FALSE)

cat("Rows:", nrow(df), "\n")
str(df)

# --- example summaries ---
# lot-wise area by crop and season
if (nrow(df) > 0) {
  agg <- aggregate(area_ha ~ crop + season, data = df, FUN = function(x) sum(as.numeric(x), na.rm = TRUE))
  print(agg)
}

File: analysis/read_data.py
# Read the synced Seed Production Log from your Google Sheet into Python.
# pip install pandas
#
# Replace SHEET_ID with your sheet's id (the long string in its URL:
# https://docs.google.com/spreadsheets/d/SHEET_ID/edit )

import pandas as pd

SHEET_ID = "1bfQs9LS-zign0eLmWYA8JfJ3LKBcMF7PDRLgtbMbZKU"
url = f"https://docs.google.com/spreadsheets/d/{SHEET_ID}/export?format=csv"

df = pd.read_csv(url)
print("Rows:", len(df))
print(df.head())

# --- example summary: area by crop and season ---
if len(df):
    print(df.groupby(["crop", "season"])["area_ha"].sum())
